packages/frontend/core/src/modules/workbench/services/desktop-state-synchronizer.ts

```diff
@@ -80,10 +64,8 @@ export class DesktopStateSynchronizer extends Service {
 
     // sync workbench state with main process
     // also fill tab view meta with title & moduleName
-    this.workspaceProperties.workspace.engine.rootDocState$
+    workbench.views$
```

Comment:
This can be simplified:

```
LiveData.compute((get) => {
  return get(workbench.views$).map((view) => ({
    title: get(view.title$),
    location: get(view.location$),
    ....
  }))
}).subscribe(
  (data) => updateWorkbenchMeta(data)
)
```